rest/route/middleware.go

```diff
@@ -446,6 +447,46 @@ func updateHostAccessTime(ctx context.Context, h *host.Host) {
 	}
 }
 
+type sageMiddleware struct{}
+
+// NewSageMiddleware returns a middleware that verifies the request
+// is coming from Sage.
+func NewSageMiddleware() gimlet.Middleware {
+	return &sageMiddleware{}
+}
+
+func (m *sageMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
```

Comment:
Maybe I'm missing something, but is there something here that's supposed to check specifically for the Sage service user? Otherwise, it seems like any user would pass auth.

Reply:
[This](https://github.com/evergreen-ci/evergreen/pull/9266/files#diff-a43d6cf099c73acfbb62513db36235b003a87fa37a9e2c303be73d31c75ba3e7R461) line ensures it is being called by the sage service user